start/manage/update.php

```diff
@@ -17,57 +13,117 @@
+    } elseif ($ship->getJediFactor() < 0){
+        $errmessages[] = "Jedi Factor can't be less than zero";
+    } elseif (is_int($ship->getJediFactor() == false)){
+        $errmessages[] = "Jedi Factor must be a number";
+    }
+
+    if (empty(trim($ship->getStrength()))){
+        $errmessages[] = "Please enter ship strength";
+    } elseif ($ship->getStrength() < 0) {
+        $errmessages[] = "Strength can't be less than zero";
+    } elseif (is_int($ship->getstrength() == false)){
+        $errmessages[] = "Strength must be a number";
+    }
+
+    if (empty(trim($ship->getDescription()))){
+        $errmessages[] = "Please enter ship description";
+    }
+
+    if ($errmessages == []){
+
+        $shipStorage = $container->getShipStorage();
+        $shipStorage->updateShip($ship);
+
+        header('location: index.php');
+    }
+
 }
 
 ?>
 
 <div class='container'>
     <a href="show.php?id=<?php echo $id?>"> Back to previous page </a>
 
+    <?php if ($ship === null): ?>
+        <h1> Ship Not Available </h1>
+    <?php else: ?>
+
     <h2>Update <?php echo $ship->getName();?> Ship Details:  </h2>
     
-    <br>
+    <div class="col-lg-3">
+        <ul class="list-group">
+            <?php foreach($errmessages as $errmessage):  ?>
+                <li class="list-group-item list-group-item-danger"><?php echo $errmessage ?></li>
+            <?php endforeach;  ?>
+        </ul>
+    </div>
 
-    <form action="update.php" method="POST">
-
-            <div>
-                <label for="shipname">Update Ship Name:</label><br>
-                <input class="form-control" type="text" name="ship" id="ship" />
-            </div>
-
-            <div>
-                <label for="weaponPower">Update Weapon Power:</label>
-                <input class="form-control" type="number" name="weaponPower" id="weaponPower" />
-            </div>  
-
-            <div>
-                <label for="jediFactor">Update Jedi Factor:</label>
-                <input class="form-control" type="number" name="jediFactor" id="jediFactor" />
-            </div>
-
-            <div>
-                <label for="strength">Update Strength:</label>
-                <input class="form-control" type="number" name="strength" id="strength" />
-            </div>
-
-            <div>
-                <label for="team">Update Ship Team:</label>
-                <input class="form-control" type="text" name="team" id="team" />
-            </div>
-
-            <div> 
-                <label for="team">Update Ship Description:</label>
-                <textarea class="form-control" name="description" id="description"></textarea>
-            </div>
-
-            <br>
-            <div class='text-center'>
-                <button type="submit" class="btn btn-success">Submit</button>
-            </div>
-        
-    </form>
+    <br>
+    
+    <div class="col-lg-12">
+        <form action="update.php?id=<?php echo $ship->getId();?>" method="POST">
```

Comment:
All of the form fields here are still doing a check for the old variables you were using. I think you can remove all of them now.